contracts/src/libraries/LPMath.sol

```diff
@@ -28,6 +28,103 @@ library LPMath {
+        uint256 timeStretch;
+        uint256 longsOutstanding;
+        uint256 longAverageTimeRemaining;
+        uint256 shortsOutstanding;
+        uint256 shortAverageTimeRemaining;
+    }
+
+    struct DistributeExcessIdleParams {
+        PresentValueParams presentValueParams;
+        uint256 startingPresentValue;
+        uint256 activeLpTotalSupply;
+        uint256 withdrawalSharesTotalSupply;
+        uint256 idle;
+        int256 netCurveTrade;
+        uint256 originalShareReserves;
+        int256 originalShareAdjustment;
+        uint256 originalBondReserves;
+    }
+
+    /// @dev Calculates the initial reserves. We solve for the initial reserves
+    ///      by solving the following equations simultaneously:
+    ///
+    ///      (1) c * z = c * z_e + p_target * y
+    ///
+    ///      (2) p_target = ((mu * z_e) / y) ** t_s
+    ///
+    ///      where p_target is the target spot price implied by the target spot
+    ///      rate.
+    /// @param _shareAmount The amount of shares used to initialize the pool.
+    /// @param _vaultSharePrice The vault share price.
+    /// @param _initialVaultSharePrice The initial vault share price.
+    /// @param _targetApr The target rate.
+    /// @param _positionDuration The position duration.
+    /// @param _timeStretch The time stretch.
+    /// @return shareReserves The initial share reserves.
+    /// @return shareAdjustment The initial share adjustment.
+    /// @return bondReserves The initial bond reserves.
+    function calculateInitialReserves(
+        uint256 _shareAmount,
+        uint256 _vaultSharePrice,
+        uint256 _initialVaultSharePrice,
+        uint256 _targetApr,
+        uint256 _positionDuration,
+        uint256 _timeStretch
+    )
+        internal
+        pure
+        returns (
+            uint256 shareReserves,
+            int256 shareAdjustment,
+            uint256 bondReserves
+        )
+    {
+        // NOTE: Round down to underestimate the initial bond reserves.
+        //
+        // Normalize the time to maturity to fractions of a year since the
+        // provided rate is an APR.
+        uint256 t = _positionDuration.divDown(365 days);
+
+        // NOTE: Round up to underestimate the initial bond reserves.
+        //
+        // Calculate the target price implied by the target rate.
+        uint256 targetPrice = ONE.divUp(ONE + _targetApr.mulDown(t));
+
+        // The share reserves is just the share amount since we are initializing
+        // the pool.
+        shareReserves = _shareAmount;
+
+        // NOTE: Round down to underestimate the initial bond reserves.
+        //
+        // Calculate the initial bond reserves. This is given by:
+        //
+        // y = (mu * c * z) / (c * p_target ** (1 / t_s) + mu * p_target)
+        bondReserves = _initialVaultSharePrice.mulDivDown(
+            _vaultSharePrice.mulDown(shareReserves),
+            _vaultSharePrice.mulUp(targetPrice.pow(ONE.divDown(_timeStretch))) +
```

Comment:
I think that rounding up for the `1/t_s` would result in a larger denominator, which would further underestimate the bond reserves.

```suggestion
            _vaultSharePrice.mulUp(targetPrice.pow(ONE.divUp(_timeStretch))) +
```

Reply:
In this case, it's actually the opposite. `targetPrice` is less than one (we know this because the target rate is a `uint256` and can thus never be less than 0). With this in mind, exponentiating `targetPrice` by a smaller exponent will result in a larger value. 

Think of this example. We have that $\frac{1}{2} = \sqrt{\frac{1}{4}}$ since $\frac{1}{2} \cdot \frac{1}{2} = \frac{1}{4}$. We also have that $\frac{1}{2} = \frac{1}{4}^{\frac{1}{2}}  > \frac{1}{4}^{1} = \frac{1}{4}$.

Reply:
Really appreciate the explanation, makes total sense.